### ISSUE TITLE
MeasureObjectIntensityDistribution calculates intensity zernikes even when not asked to.

### ISSUE THREAD

**Issue by bethac07** [CellProfiler/CellProfiler#2193]
Running an analysis pipeline on AWS on the stable 2.2.0 I got the following error:
```
Wed Aug 10 18:48:16 2016: Image # 19, module MeasureObjectNeighbors # 34: 3.55 sec
Error detected during run of module MeasureObjectIntensityDistribution
Traceback (most recent call last):
  File "/usr/local/src/CellProfiler/cellprofiler/pipeline.py", line 1819, in run_with_yield
    self.run_module(module, workspace)
  File "/usr/local/src/CellProfiler/cellprofiler/pipeline.py", line 2067, in run_module
    module.run(workspace)
  File "/usr/local/src/CellProfiler/cellprofiler/modules/measureobjectintensitydistribution.py", line 438, in run
    self.calculate_zernikes(workspace)
  File "/usr/local/src/CellProfiler/cellprofiler/modules/measureobjectintensitydistribution.py", line 786, in calculate_zernikes
    z_ = z[mask, :]
MemoryError
Wed Aug 10 18:48:19 2016: Image # 19, module MeasureObjectIntensityDistribution # 35: 63.51 sec
```

I can't post the full pipeline in this case but the relevant module settings from the cppipe are below:
```
MeasureObjectIntensityDistribution:[module_num:35|svn_version:\'Unknown\'|variable_revision_number:5|show_window:False|notes:\x5B\x5D|batch_state:array(\x5B\x5D, dtype=uint8)|enabled:True|wants_pause:False]
    Hidden:2
    Hidden:3
    Hidden:1
    Hidden:0
    Calculate intensity Zernikes?:None
    Maximum zernike moment:9
    Select an image to measure:Channel2
    Select an image to measure:Channel3
    Select objects to measure:Cells
    Object to use as center?:These objects
    Select objects to use as centers:None
    Select objects to measure:Cytoplasm
    Object to use as center?:These objects
    Select objects to use as centers:None
    Select objects to measure:Nuclei
    Object to use as center?:These objects
    Select objects to use as centers:None
    Scale the bins?:Yes
    Number of bins:4
    Maximum radius:100
```

My pipeline is crashing due to memory errors on seemingly 75+% of the images so I'm highly suspicious this is why.

**Comment by tomgreen66** [comment]
This looks very similar to a problem we are experiencing but we got a bit further possibly:
```
Error detected during run of module MeasureObjectIntensityDistribution
Traceback (most recent call last):
  File "/software/tools/pyenv/versions/2.7.9-cellprofiler/src/cellprofiler/cellprofiler/pipeline.py", line 1819, in run_with_yield
    self.run_module(module, workspace)
  File "/software/tools/pyenv/versions/2.7.9-cellprofiler/src/cellprofiler/cellprofiler/pipeline.py", line 2067, in run_module
    module.run(workspace)
  File "/software/tools/pyenv/versions/2.7.9-cellprofiler/src/cellprofiler/cellprofiler/modules/measureobjectintensitydistribution.py", line 438, in run
    self.calculate_zernikes(workspace)
  File "/software/tools/pyenv/versions/2.7.9-cellprofiler/src/cellprofiler/cellprofiler/modules/measureobjectintensitydistribution.py", line 775, in calculate_zernikes
    yx[:, 1], yx[:, 0], zernike_indexes)
  File "/software/tools/pyenv/versions/2.7.9-cellprofiler/lib/python2.7/site-packages/centrosome/zernike.py", line 69, in construct_zernike_polynomials
    result = np.zeros(list(mask.shape) + [nzernikes], np.complex)
MemoryError
Fri Aug  5 17:02:33 2016: Image # 1, module MeasureObjectIntensityDistribution # 14: 1765.02 sec
```
It would be interesting to see the scaling of memory requirement with image size since it works with smaller images.

**Comment by tomgreen66** [comment]
Having looked at the code I have spotted a possible issue.  See fix in c63f783 - will be testing it to see if this works since it does look strange to since there is no `__nonzero__` or `__len__` defined for the Choice class from what I can see.  From Python documentation you would have to have the following in use:

>  object.__nonzero__(self)
Called to implement truth value testing and the built-in operation bool(); should return False or True, or their integer equivalents 0 or 1. When this method is not defined, __len__() is called, if it is defined, and the object is considered true if its result is nonzero. If a class defines neither __len__() nor __nonzero__(), all its instances are considered true.

**Comment by mcquin** [comment]
Resolved by https://github.com/CellProfiler/CellProfiler/pull/2296
